src/vm/VMrunner.ts

```diff
@@ -555,13 +556,11 @@ export class VMProgramRunner extends JDClient {
     }
     
     // timers
-    async wakeSleeperAsync(handler: VMHandlerRunner, ms: number) {
-        console.log("wakeSleeper")
+    async sleepAsync(handler: VMHandlerRunner, ms: number) {  
         await this._sleepMutex.acquire(async () => {
-            let fireOnce = () => { 
+            let id = setTimeout(() => { 
                 this.emit(VM_WAKE_SLEEPER, handler)
```

Comment:
the runner may have been stopped by the time you reach this handler

Reply:
I have logic on the other side for this.